packages/@n8n/utils/src/retry.ts

```diff
@@ -0,0 +1,40 @@
+type RetryFn = () => boolean | Promise<boolean>;
+
+/**
+ * A utility that retries a function every `interval` milliseconds
+ * until the function returns true or the maximum number of retries is reached.
+ *
+ * @param fn - A function that returns a boolean or a Promise resolving to a boolean.
+ * @param interval - The time interval (in milliseconds) between each retry. Defaults to 1000.
+ * @param maxRetries - The maximum number of retry attempts. Defaults to 3.
+ * @returns {Promise<boolean>} - A promise that resolves to:
+ *   - true: If the function returns true before reaching maxRetries.
+ *   - false: If the function never returns true or if an error occurs.
+ */
+export async function retry(
+	fn: RetryFn,
+	interval: number = 1000,
+	maxRetries: number = 3,
+): Promise<boolean> {
+	let attempt = 0;
+
+	while (attempt < maxRetries) {
+		attempt++;
+		try {
+			const result = await fn();
+			if (result) {
+				return true;
+			}
+		} catch (error) {
+			console.error('Error during retry:', error);
+			throw error;
+		}
+
+		// Wait for the specified interval before the next attempt, if any attempts remain.
+		if (attempt < maxRetries) {
+			await new Promise<void>((resolve) => setTimeout(resolve, interval));
```

Comment:
instead of a constant value for `interval`, we could use `attempt` to gradually increase the duration between retries.

Reply:
Added `backoff` strategy as `linear | exponential | null` to gradually increase retries.